### PR TITLE
Show tables of summary statistics in summary section of cs2

### PR DESCRIPTION
**Pull request by chroetz**
Two functions are added:
* `calcTimeSeriesStats()` takes a quitte object (loaded from a mif file) and calculates given statistics of the time series of given variables, e.g., max, hitting time of max, hitting time of 0.
* `showStatsTable()` creates a displayable table from the results of `calcTimeSeriesStats()` 

The functions are used to add tables of the periods of max/min CO2/GHG emissions, first period of net 0 CO2/GHG emissions, and max period of warming to the end of the summary section of compareScenarios2.

The results looks like this:
t1
t2
t3

### CODE REVIEW

File: R/calcTimeSeriesStats.R [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  stopifnot(is.list(stats))
+  stopifnot(length(stats) > 0)
+  stopifnot(!is.null(names(stats)))
+  stopifnot(is.character(var))
+  stopifnot(is.numeric(from) && is.numeric(to))
+  stopifnot(length(from) == 1 && length(to) == 1)
+
+  idCols <- c("model", "scenario", "region", "variable")
+  data %>%
+    filter(.data$variable %in% .env$var, .data$period >= .env$from, .data$period <= .env$to) ->
```
Comment (pfuehrlich-pik) [inline R/calcTimeSeriesStats.R:47]: The linter should flag `->` as undesirable operator, please always use `<-` even when using pipes.

**Review by pfuehrlich-pik — CHANGES_REQUESTED** [review]

**Comment by pfuehrlich-pik** [comment]
The tables look super professional and nice, thanks for adding them! :sunglasses:

File: DESCRIPTION [hunk 2]
```diff
@@ -19,6 +19,7 @@ Imports:
     ggplot2,
     highcharter,
     iamc,
+    kableExtra,
```
[the remark this answers is not in the document]
Comment (chroetz) [reply DESCRIPTION:22]: Thanks, good point! How should such a check look like?

File: R/calcTimeSeriesStats.R (hunk 1 above)
Comment (chroetz) [reply R/calcTimeSeriesStats.R]: true! I replaced the assignment operators accordingly

File: R/calcTimeSeriesStats.R [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#' @importFrom rlang .env .data
+#' @export
+calcTimeSeriesStats <- function(
+    data,
+    var,
+    stats,
+    from = -Inf,
+    to = 2100) {
+
+  stopifnot(quitte::is.quitte(data))
```
[the remark this answers is not in the document]
Comment (chroetz) [reply R/calcTimeSeriesStats.R:37]: Ah thanks, I didn't know that you can put multiple checks in one `stopifnot()`.

I feel like most of the time the code speaks quite well for itself. So, yes, `Error: data is not a quitte object`  is better than `Error: quitte::is.quitte(data) is not TRUE`. But only a tiny tiny bit. Furthermore, the code is typically more precise (or at least not less precise, unless you do calculations outside the `stopifnot()` statement). So, I'm wondering whether it is worth the effort to comment all `stopifnot()` calls... What do you think?

**Review by pfuehrlich-pik — APPROVED** [review]
Looks good!

File: DESCRIPTION (hunk 2 above)
[the remark this answers is not in the document]
Comment (pfuehrlich-pik) [reply DESCRIPTION:22]: Take look [here](https://r-pkgs.org/dependencies.html#guarding-the-use-of-a-suggested-package)

[the remark this answers is not in the document]
Comment (chroetz) [reply DESCRIPTION]: thanks

File: R/calcTimeSeriesStats.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (pfuehrlich-pik) [reply R/calcTimeSeriesStats.R:37]: I agree, in most cases seeing which code failed is fine, I just wanted to point out the option.

File: inst/markdown/compareScenarios2/cs2_main.Rmd [hunk 4]
```diff
@@ -207,11 +208,11 @@ rm(dataScen, dataScenNested, dataHist)
 data %>%
   # In the variable names, replace `$dot$` by `.`.
   mutate(
-    variable = remind2::substituePlaceholder(variable)) %>%
+    variable = substituePlaceholder(variable)) %>%
   # In the variable names, replace `|+|`, `|++|`, `|+++|`, ... by `|`.
   mutate(
     varplus = as.character(variable),
-    variable = remind2::deletePlus(variable)) ->
+    variable = deletePlus(variable)) ->
```
Comment (pfuehrlich-pik) [inline inst/markdown/compareScenarios2/cs2_main.Rmd:215]: Here's another `->` which should be replaced

**Review by pfuehrlich-pik — APPROVED** [review]

File: inst/markdown/compareScenarios2/cs2_main.Rmd (hunk 4 above)
Comment (chroetz) [reply inst/markdown/compareScenarios2/cs2_main.Rmd:215]: The cs2-Rmd-files do not really follow the code style conventions yet. There is much more to change there. I will do that in a different PR

**Pull request merged**